conntable.go

```diff
@@ -71,20 +74,42 @@ func (t *connTable) Register(ck ckey, matchedRule *Rule, srcIP, srcPort string,
 			TargetPort: targetPort,
 			//TargetIP:   targetIP,
 		}
+
 	}
 }
 
-func (t *connTable) FlushOlderThan(s time.Duration) {
+func (t *connTable) FlushOlderOnes() {
 	t.mtx.Lock()
 	defer t.mtx.Unlock()
 
-	threshold := time.Now().Add(-1 * s)
+	for minutes := 5; minutes >= 0; minutes -= 1 {
 
-	for ck, md := range t.table {
-		if md.Added.Before(threshold) {
-			delete(t.table, ck)
+		duration := time.Duration(minutes) * time.Minute
+		threshold := time.Now().Add(-1 * duration)
+
+		for ck, md := range t.table {
+			if md.Added.Before(threshold) {
+				delete(t.table, ck)
+			}
 		}
+
+		if t.maxConns > uint32(len(t.table)) {
```

Comment:
minor comment: I usually find it easier to understand/parse if I put the `dynamic` part of the equation on the left hand side. So it would be:

`if len(t.table) < t.softLimit { break }`